hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -125,12 +125,14 @@ class BaseSocket : public hazelcast::client::socket
             add_invocation_to_map(connection, invocation, message);
 
             auto& datas = message->get_buffer();
-            std::vector<boost::asio::const_buffer> buffers;
-            buffers.reserve(datas.size());
+
+            struct outbox_entry_t outbox_entry;
+            outbox_entry.buffers.reserve(datas.size());
             for (const auto& data : datas) {
-                buffers.emplace_back(boost::asio::buffer(data));
+                outbox_entry.buffers.emplace_back(boost::asio::buffer(data));
             }
-            this->outbox_.push_back(buffers);
+            outbox_entry.invocation = invocation;
+            this->outbox_.push_back( outbox_entry );
```

Comment:
`outbox_entry` should be moved to eliminate copy cost.

```suggestion
            this->outbox_.push_back(std::move(outbox_entry));
```

Reply:
fixed